src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+}
+
+
+/**
+ * @brief Returns the azimuthal angle value.
+ * @param azim index of the azimthal angle of interest
+ * @return the value of the azimuthal angle
+ */
+double Quadrature::getPhi(int azim) {
+
+  if (azim < 0 || azim >= _num_azim)
+    log_printf(ERROR, "Attempted to retrieve theta for "
+               "azim angle = %d when only %d azim angles are "
+               "defined", azim, _num_azim);
+
+  else if (_phis == NULL)
+    log_printf(ERROR, "Attempted to retrieve phi for "
+               "azim angle = %d but the phis have not been "
+               "initialized", azim);
+
+  return _phis[azim];
+}
+
+
+/**
+ * @brief Returns the azimuthal angle weight value for a particular azimuthal angle.
+ * @param azim index of the azimuthal angle of interest
+ * @return the weight for an azimuthal angle
+ */
+FP_PRECISION Quadrature::getAzimWeight(int azim) {
+
+  if (azim < 0 || azim >= _num_azim)
+    log_printf(ERROR, "Attempted to retrieve the azimuthal weight for "
+               "azimuthal angle = %d but only %d azimuthal angles "
+               "are defined", azim, _num_azim);
+
+  else if (_azim_weights == NULL)
+    log_printf(ERROR, "Attempted to retrieve weight for azimuthal angle = %d "
+               "but the azimuthal weights have not been initialized", azim);
+
+  return _azim_weights[azim];
+}
+
+
+/**
+ * @brief Returns the polar weight for a particular azimuthal and polar angle.
+ * @param azim index of the azimthal angle of interest
+ * @param polar index of the polar angle of interest
+ * @return the value of the polar weight for this azimuthal and polar angle
+ */
+FP_PRECISION Quadrature::getPolarWeight(int azim, int polar) {
+
+  if (polar < 0 || polar >= _num_polar)
+    log_printf(ERROR, "Attempted to retrieve polar weight for polar angle = "
+               "%d and azim angle = %d when only %d polar angles are "
+               "defined", polar, azim, _num_polar);
+
+  else if (azim < 0 || azim >= _num_azim)
+    log_printf(ERROR, "Attempted to retrieve polar weight for polar angle = "
+               "%d and azim angle = %d when only %d azim angles are "
+               "defined", polar, azim, _num_azim);
+
+  else if (_polar_weights == NULL)
+    log_printf(ERROR, "Attempted to retrieve polar weight for polar angle = %d "
+               "and azim angle = %d but the thetas have not been "
+               "initialized", polar, azim);
+
+  return _polar_weights[azim][polar];
+}
+
+
+/**
+ * @brief Returns the total weight for Tracks with the given azimuthal and
+ *        polar indexes
+ * @details Angular weights are multiplied by Track spcings
+ * @param azim index of the azimuthal angle of interest
+ * @param polar index of the polar angle of interest
+ * @return the total weight of each Track with the given indexes
+ */
+FP_PRECISION Quadrature::getWeight(int azim, int polar) {
+
+  if (polar < 0 || polar >= _num_polar)
+    log_printf(ERROR, "Attempted to retrieve the multiple for polar angle = "
```

Comment:
This error message won't make much sense anymore when it refers to "multiple" since this method is now returning the "weight." Perhaps just reconstruct the error messages to align with the method signature.